bazel/protobuf.bzl

```diff
@@ -21,10 +21,21 @@ def _path_ignoring_repository(f):
         # before "external/workspace", so we need to add the starting index of "external/workspace"
         return f.path[f.path.find(f.owner.workspace_root) + len(f.owner.workspace_root) + 1:]
 
+def _common_suffix(a, b):
+    matching_parts = []
+    for (x, y) in zip(reversed(a), reversed(b)):
+        if x != y:
+            break
+        matching_parts.append(x)
+    return reversed(matching_parts)
+
 def _package_relative_path(ctx, file):
     # Remove the path prefix common to both the BUILD file that defines the
     # package and the input file.
-    path_parts = file.short_path.split("/")
+
+    # file.short_path includes the root and file.path is relative to the root. We remove the root
+    # and keep only the path relative to the workspace root.
```

Comment:
Can you include some examples of both the top-level package relative path and the intermediate working state? I think it would help as this is now quite involved path manipulation.

Should we have a regression test for the Envoy-side issue (assuming its not a lot of work to put in place)?

Reply:
I can start by documenting them here; let me know if you want more in the actual code. From some print statements I had added while working on this:

```
# package @envoy_api_canonical//envoy/annotations
ctx.build_file_path = envoy_api_canonical/envoy/annotations/BUILD
file.short_path = ../envoy_api_canonical/envoy/annotations/resource.proto
file.path = external/envoy_api_canonical/envoy/annotations/resource.proto
file.dirname = external/envoy_api_canonical/envoy/annotations

# package @opentelemetry_proto//
ctx.build_file_path = opentelemetry_proto/BUILD.bazel
file.short_path = ../opentelemetry_proto/opentelemetry/proto/collector/logs/v1/logs_service.proto
file.path = external/opentelemetry_proto/opentelemetry/proto/collector/logs/v1/logs_service.proto
file.dirname = external/opentelemetry_proto/opentelemetry/proto/collector/logs/v1
```

These fields are (mostly) documented here: https://docs.bazel.build/versions/master/skylark/lib/File.html. As for a regression test, I'm not sure yet how to create one without setting up a separate Bazel workspace and importing PGV. That's doable and probably not terrible, just unclear whether it's worth including here. WDYT?

Reply:
Yeah, I think if you include those in a comment it would be great, thanks, it makes it clear to me what is going on if you also annotate the second example and explain the BUILD is in a different place to the proto.

Re: test, up to you, but maybe file an issue at least around this.